site/content/kapp-controller/docs/develop/kctrl-package-authoring.md

```diff
@@ -0,0 +1,543 @@
+```
+Once the repository is added successfully we should be able to see our packages on the cluster.
+```bash
+$ kctrl package available list
+Target cluster 'https://127.0.0.1:49841' (nodes: minikube)
+
+Available summarized packages in namespace 'default'
+
+Name                    Display name  
+certmanager.carvel.dev  certmanager  
+simple-app.carvel.dev   simple-app  
+
+Succeeded
+```
+Great! We have now bundled and published two of our packages together as a PackageRepository.
+
+## Testing packages locally
+`kctrl` enables users authoring Apps and Packages using `kapp-controller`'s APIs to test their resources effectively.
+Let us test our `simple-app` package locally. 
+
+This can be done by using the `dev deploy` command, and it does not require `kapp-controller` to be installed
+on the cluster. The fetch and template steps are done locally on the host running the command before
+using `kapp` to deploy the resources on the cluster.
+
+Let us take a look at the `package-resources.yml` file generated by `kctrl` while initialising the package.
+```yaml
+#package-resources.yml
+
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: Package
+metadata:
+  creationTimestamp: null
+  name: simple-app.carvel.dev.0.0.0
+spec:
+  refName: simple-app.carvel.dev
+  releasedAt: null
+  template:
+    spec:
+      deploy:
+      - kapp: {}
+      fetch:
+      - git: {}
+      template:
+      - ytt:
+          paths:
+          - config
+      - kbld: {}
+  valuesSchema:
+    openAPIv3: null
+  version: 0.0.0
+
+---
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: PackageMetadata
+metadata:
+  creationTimestamp: null
+  name: simple-app.carvel.dev
+spec:
+  displayName: simple-app
+  longDescription: simple-app.carvel.dev
+  shortDescription: simple-app.carvel.dev
+
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageInstall
+metadata:
+  annotations:
+    kctrl.carvel.dev/local-fetch-0: .
+  creationTimestamp: null
+  name: simple-app
+spec:
+  packageRef:
+    refName: simple-app.carvel.dev
+    versionSelection:
+      constraints: 0.0.0
+  serviceAccountName: simple-app-sa
+status:
+  conditions: null
+  friendlyDescription: ""
+  observedGeneration: 0
+```
+The `dev deploy` command will process these resources just like they would on the cluster.
+However, we would like to build an image from source and use the configuration in the root of the directory.
+We use the `--local` and `--build` flags to indicate this.
+
+`kctrl` needs to be informed where it can find the folder `config`. This is done by using the annotation 
+`kctrl.carvel.dev/local-fetch-0: .` on the PackageInstall resource. It tells `dev deploy` that
+the files `kapp-controller` would otherwise fetch, is available in the root of the project.
+
+Let's build and deploy from source.
+```bash
+$ kctrl dev deploy -f package-resources.yml
```

Comment:
If we are building the binary as well, we need to add `-b` and `-l` flags. Otherwise we need only `-l` flag.